File: ExampleScutiObjC/Pages/CustomScutiButtonViewController.m
//
//  CustomScutiButtonViewController.m
//  ExampleScutiObjC
//
//  Created by mac on 26/09/2023.
//

#import "CustomScutiButtonViewController.h"
@import ScutiSDKSwift;

@interface CustomScutiButtonViewController () <ScutiSDKManagerDelegate>
@property (weak, nonatomic) IBOutlet UILabel *lblStoreReady;
@property (weak, nonatomic) IBOutlet UILabel *lblUserAuthenticated;
@property (weak, nonatomic) IBOutlet UILabel *lblCntProducts;
@property (weak, nonatomic) IBOutlet UILabel *lblCntRewards;

@property (weak, nonatomic) IBOutlet UILabel *lblButtonNew;
@property (weak, nonatomic) IBOutlet UILabel *lblButtonNewRewards;

@end

@implementation CustomScutiButtonViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    ScutiSDKManager.shared.delegate = self;
    if (ScutiSDKManager.shared.scutiEventsObjC.isStoreReady) {
        self.lblStoreReady.text = @"Store Ready : ON";
        self.lblStoreReady.textColor = UIColor.greenColor;
    }
    if (ScutiSDKManager.shared.scutiEventsObjC.userToken != nil) {
        self.lblUserAuthenticated.text = @"User authenticated : YES";
        self.lblUserAuthenticated.textColor = UIColor.greenColor;
    }
    self.lblCntProducts.text = [NSString stringWithFormat:@"New Products : %ld", (long)ScutiSDKManager.shared.scutiEventsObjC.cntNewProducts];
    if (ScutiSDKManager.shared.scutiEventsObjC.cntNewProducts > 0) {
        [self.lblButtonNew setHidden:false];
        self.lblButtonNew.text = [NSString stringWithFormat:@"New: %ld", (long)ScutiSDKManager.shared.scutiEventsObjC.cntNewProducts];
    }
    self.lblCntRewards.text = [NSString stringWithFormat:@"New Rewards : %ld", (long)ScutiSDKManager.shared.scutiEventsObjC.cntRewards];
    if (ScutiSDKManager.shared.scutiEventsObjC.cntRewards > 0) {
        [self.lblButtonNewRewards setHidden:false];
        self.lblButtonNewRewards.text = [NSString stringWithFormat:@"%ld", (long)ScutiSDKManager.shared.scutiEventsObjC.cntRewards];
    }
}
- (IBAction)onStore:(id)sender {
    [ScutiSDKManager.shared showScutiWebViewWithViewController:self];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/
#pragma mark - ScutiSDKManagerDelegate

- (void)onBackToGame {
    
}

- (void)onErrorOccurredWithError:(NSError * _Nonnull)error {
    
}

- (void)onLogout {
    self.lblUserAuthenticated.text = @"User authenticated : NO";
    self.lblUserAuthenticated.textColor = UIColor.redColor;
}

- (void)onNewProductsWithCntProducts:(NSInteger)cntProducts {
    self.lblCntProducts.text = [NSString stringWithFormat:@"New Products : %ld", (long)cntProducts];
    [self.lblButtonNew setHidden:cntProducts == 0];
    self.lblButtonNew.text = [NSString stringWithFormat:@"New: %ld", (long)cntProducts];
}

- (void)onNewRewardsWithCntRewards:(NSInteger)cntRewards {
    self.lblCntRewards.text = [NSString stringWithFormat:@"New Rewards : %ld", (long)cntRewards];
    [self.lblButtonNewRewards setHidden:cntRewards == 0];
    self.lblButtonNewRewards.text = [NSString stringWithFormat:@"%ld", (long)cntRewards];
}

- (void)onScutiButtonClicked {
    
}

- (void)onScutiExchangeWithExchange:(ScutiExchangeClass * _Nonnull)exchange {
    
}

- (void)onStoreReady {
    self.lblStoreReady.text = @"Store Ready : ON";
    self.lblStoreReady.textColor = UIColor.greenColor;
}

- (void)onUserTokenWithUserToken:(NSString * _Nonnull)userToken {
    self.lblUserAuthenticated.text = @"User authenticated : YES";
    self.lblUserAuthenticated.textColor = UIColor.greenColor;
}

@end

File: Scuti/Pages/CustomScutiButtonViewController.swift
//
//  CustomScutiButtonViewController.swift
//  ExampleScutiSwift
//
//  Created by mac on 21/09/2023.
//

import UIKit
import ScutiSDKSwift

class CustomScutiButtonViewController: UIViewController {

    @IBOutlet weak var lblStoreReady: UILabel!
    @IBOutlet weak var lblUserAuthenticated: UILabel!
    @IBOutlet weak var lblCntProducts: UILabel!
    @IBOutlet weak var lblCntRewards: UILabel!

    @IBOutlet weak var lblButtonNew: UILabel!
    @IBOutlet weak var lblButtonNewRewards: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        ScutiSDKManager.shared.delegate = self
        if ScutiSDKManager.shared.scutiEvents.isStoreReady {
            lblStoreReady.text = "Store Ready : ON"
            lblStoreReady.textColor = .green
        }
        if ScutiSDKManager.shared.scutiEvents.userToken != nil {
            lblUserAuthenticated.text = "User authenticated : YES"
            lblUserAuthenticated.textColor = .green
        }
        lblCntProducts.text = "New Products : \(ScutiSDKManager.shared.scutiEvents.cntNewProducts)"
        if ScutiSDKManager.shared.scutiEvents.cntNewProducts > 0 {
            lblButtonNew.isHidden = false
            lblButtonNew.text = "New: \(ScutiSDKManager.shared.scutiEvents.cntNewProducts)"
        }
        lblCntRewards.text = "New Rewards : \(ScutiSDKManager.shared.scutiEvents.cntRewards)"
        if ScutiSDKManager.shared.scutiEvents.cntRewards > 0 {
            lblButtonNewRewards.isHidden = false
            lblButtonNewRewards.text = "\(ScutiSDKManager.shared.scutiEvents.cntRewards)"
        }
    }

    @IBAction func onStore(_ sender: Any) {
        ScutiSDKManager.shared.showScutiWebView(viewController: self)
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
extension CustomScutiButtonViewController: ScutiSDKManagerDelegate {
    func onStoreReady() {
        lblStoreReady.text = "Store Ready : ON"
        lblStoreReady.textColor = .green
    }
    func onLogout() {
        lblUserAuthenticated.text = "User authenticated : NO"
        lblUserAuthenticated.textColor = .red
    }
    func onNewProducts(cntProducts: Int) {
        lblCntProducts.text = "New Products : \(cntProducts)"
        lblButtonNew.isHidden = cntProducts == 0
        lblButtonNew.text = "New: \(cntProducts)"
    }
    func onNewRewards(cntRewards: Int) {
        lblCntRewards.text = "New Rewards : \(cntRewards)"
        lblButtonNewRewards.isHidden = cntRewards == 0
        lblButtonNewRewards.text = "\(cntRewards)"
    }
    func onUserToken(userToken: String) {
        lblUserAuthenticated.text = "User authenticated : YES"
        lblUserAuthenticated.textColor = .green
    }
}
